packages/server/src/handlers/chat.ts

```diff
@@ -40,3 +45,20 @@ export const handleChat = async (
     });
   }
 };
+
+function prepareUserInput(messages: Message[]) {
+  let userInput = messages[messages.length - 1]?.content || "";
+
+  const lastArtifact = getLastArtifactFromMessages(messages);
+  if (lastArtifact?.type === "code") {
+    const { data } = lastArtifact as CodeArtifact;
+    userInput += `\n\nHere's the current code:\nFile: ${data.file_name}\n\n${data.code}`;
+  }
```

Comment:
from my understanding, the artifact generator tool needs this context to generate an updated artifact - so this prompt should come ideally from the artifact generator.

we can't serialize a function in the artifact, but we can use `PromptTemplate`, so this more generic approach should be possible:

```suggestion
  if (lastArtifact) {
    const prompt = new PromptTemplate({template: lastArtifact.prompt});
    userInput += "\n\n" + prompt.format(lastArtifact.data);
  }
```

Reply:
that code would be even better because then the artifactGenerator tool will have the prompttemplate